### PR TITLE
Slice 4 PR-C — /history LOG BOOK ledger row + week aggregate (DEC-089 D5)

### PR DESCRIPTION
**Pull request by leehopper**
## Slice 4 PR-C — `/history` LOG BOOK ledger recomposition (DEC-089 D5)

Third of four Slice 4 PRs. Recomposes the `/history` (Log Book) list from cards into the Alpine **ledger row** and adds client-side week aggregates. Consumes PR-A's already-merged `isOnPlan` / `prescribedWorkoutType` fields. **Frontend-only** — no backend, codegen, or generated-client changes.

Spec: `docs/specs/slice-4-log-logbook/spec.md` §3 PR-C / DU-8 (gitignored working-tree contract).

### What changed
- **`LedgerRow`** (new) replaces the card `WorkoutLogEntry`: a hairline-separated 3-col grid — day numeral + weekday · title + status tag + note snippet · right-aligned distance/duration/pace. `WorkoutLogMetrics` and `WorkoutLogSplits` stay mounted below the grid **unchanged** (splits restyle is PR-D's scope; this keeps their behavior + coverage intact and gives PR-D its mount point).
- **Week aggregate** (`week-aggregate.helpers`): `{N.N} KM · {N} RUNS` (+ ` · {N} SKIP`). PARTIAL counts toward RUNS + the KM sum; SKIP suffix after RUNS. Carried on each `WorkoutHistoryWeekGroup`.
- **Week label** reformatted to `WEEK OF {MON} {D}` (no year).
- **On-plan title** from `prescribedWorkoutType` via `WORKOUT_TYPE_LABELS` (off-plan/legacy → `Run`) + a `· ON-PLAN` suffix from `isOnPlan`.
- **`formatLedgerDayParts`** (new, local-calendar) for the day numeral + weekday split.

### Ratified checkpoints (spec §9)
- **§9 #2** — SKIP after RUNS in the aggregate; suffix only when `skipCount > 0`.
- **§9 #4** — `WEEK OF` omits the year (bucket keys still carry it, so no correctness risk — only a rare cosmetic ambiguity for a solo user with >1yr history).

### Two a11y calls that go beyond the spec's assumptions
- **Status-tag colors.** Measured against `--background`, `--destructive` fails WCAG AA as *text* in **both** modes (4.21 dark / 3.88 light) and `--warning` fails in light (1.97). So the tags use the AA-safe text variants — **`text-positive` / `text-warning-text` / `text-danger-text`** — and all three are added as gated `check-contrast` pairs. Raw `--warning`/`--destructive` are never used as text. (The spec anticipated only `--warning` failing; the destructive-as-text failure is new, surfaced by the contrast probe.)
- **Skipped day numeral.** The mock dims it to `--alp-faint`, but that token is decorative-only / AA-failing and forbidden for essential text (spec §8, `frontend/CLAUDE.md`). The numeral uses `text-muted-foreground` instead (row-level `opacity-75` provides the dimming); the *decorative* weekday keeps faint. Same faint→muted migration Slice 2 made.

### Review & tests
Ran a pre-commit adversarial multi-agent review (correctness / spec-compliance / a11y-tokens / test-quality, each finding blind-challenged). The two logic dimensions came back clean; 5 confirmed findings — 1 a11y (the faint numeral above) + 4 test-coverage gaps (no PARTIAL-row test, unasserted `STATUS_COLOR_CLASS` colors, a no-op TZ-safety test, untested unknown-type → `Run` fallback) — were all fixed before this commit.

- `npm run build` clean · `npm run test` **1018 passed (86 files)** · `npm run lint` clean (changed files) · `npm run check-contrast` **50/50 pass** (incl. the 3 new status-tag pairs, both modes).

### Scope boundary
Splits table restyle, the `LOAD OLDER` button, and the `LOG BOOK` / `EVERY SPLIT ON RECORD` page header are deliberately **untouched** here — they are PR-D (the last PR in the train), which mounts its splits trigger inside this row.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01MKe9ogLzTZqLjNG2t2ZnGM

## Summary by CodeRabbit

* **New Features**
  * Added a ledger-style workout history view with compact daily rows.
  * Added weekly summaries showing distance, run counts, and skipped workouts.
  * Added support for workout notes, metrics, splits, status tags, and kilometer or mile units.
  * Week labels now use a shorter format without the year.

* **Bug Fixes**
  * Improved date display consistency across time zones.
  * Added contrast checks for ledger status colors to improve accessibility.

* **Tests**
  * Expanded coverage for ledger rows, weekly summaries, date formatting, and accessibility contrast.

### CODE REVIEW

File: frontend/src/app/modules/logging/history/week-aggregate.helpers.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * Formats a {@link WeekAggregate} as the ledger's week-summary line, e.g.
+ * `"15.2 km · 3 RUNS · 1 SKIP"` (source stays natural-case; the caller's
+ * `uppercase` CSS class is what renders it all-caps). The skip clause is
+ * appended AFTER the run clause only when at least one skip occurred
+ * (spec § 9 #2); a week with no skips renders just the distance + run count.
+ */
+export const formatWeekAggregate = (agg: WeekAggregate, units: PreferredUnits): string => {
+  const km = formatDistanceMeters(agg.distanceMeters, units) ?? '0.0 km'
```
Comment (leehopper) [inline frontend/src/app/modules/logging/history/week-aggregate.helpers.ts]: **🟡 [MEDIUM] Zero-distance week aggregate shows km even under a Miles preference**

formatWeekAggregate computes the distance label via formatDistanceMeters(agg.distanceMeters, units) with a nullish-coalescing fallback to a hardcoded '0.0 km' string literal. formatDistanceMeters returns null for any non-positive distance regardless of units, so a week whose runs sum to zero distance falls through to that km literal. Under a Miles preference, an all-skipped week (or a week whose runs carry zero actual distance) renders "0.0 km" instead of a mi form, inconsistent with every non-zero week and every ledger row elsewhere on the page. Reachable: computeWeekAggregate yields distanceMeters=0 for an all-skipped week (an explicitly tested aggregate shape). Untested: the zero-distance test case only exercises Kilometers; the Miles test case uses a non-zero 5000m distance.

Independently corroborated by the cross-file-impact agent and confirmed by a blind challenge at 93% confidence.

```suggestion
  const km = formatDistanceMeters(agg.distanceMeters, units) ?? `0.0 ${distanceUnitLabel(units)}`
```

**Review by leehopper — COMMENTED** [review]
### Deep Review

Found 1 issue (0 critical, 0 high, 1 medium):

1. **[bug]** Zero-distance week aggregate shows "km" even under a Miles preference — [`week-aggregate.helpers.ts:49`](https://github.com/leehopper/personal-running-coach/blob/683ae787fb45b2afd072036f14e499343084faf5/frontend/src/app/modules/logging/history/week-aggregate.helpers.ts#L48-L53)

   `formatWeekAggregate` falls back to a hardcoded `0.0 km` literal for any zero-distance week, ignoring the Miles preference. Reachable via an all-skipped week; untested under Miles.

---
Generated by deep-review | Reviewed up to: 683ae787fb45b2afd072036f14e499343084faf5

File: frontend/scripts/check-contrast.ts [hunk 2]
```diff
@@ -200,6 +200,21 @@ export const PAIRS: readonly Pair[] = [
   // token, NOT the border-accent --warning — raw --warning (--alp-amber)
   // measures only ~1.8:1 on the light card. This asserts the gated text token.
   { label: '--warning-text on --card', fg: 'warning-text', bg: 'card', threshold: 4.5 },
+  // LOG BOOK ledger-row status-tag colours (Slice 4 PR-C): Completed/Partial/
+  // Skipped tags render directly on the page --background (not a card fill),
+  // in --positive/--warning-text/--danger-text respectively — asserted here
+  // distinctly from the --card-backed --warning-text pair above. Plain
+  // --warning and --destructive are deliberately NOT substituted in: they
+  // fail AA on --background (--warning ~1.97 light; --destructive ~4.21 dark
+  // / ~3.88 light).
```
[the remark this answers is not in the document]
Comment (leehopper) [reply frontend/scripts/check-contrast.ts]: Fixed — removed the `Slice 4 PR-C` planning reference and wrapped the CSS custom-property tokens in backticks. Verified via `npm run check-contrast` (50/50 pass) and `npm run lint`.

File: frontend/src/app/modules/logging/history/ledger-row.component.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+         * skipped title below); the row-level `opacity-75` remains the
+         * primary visual dimming mechanism for the skipped state.
+         */}
+        <span
+          data-testid="workout-history-entry-day"
+          className={cn('t-numeral', isSkipped ? 'text-muted-foreground' : 'text-foreground')}
+        >
+          {dayNum}
+        </span>
+        <span className="t-data-label uppercase text-[var(--alp-faint)]">{weekday}</span>
```
[the remark this answers is not in the document]
Comment (leehopper) [reply frontend/src/app/modules/logging/history/ledger-row.component.tsx]: Fixed — the weekday now uses `text-muted-foreground` instead of the decorative, AA-failing `--alp-faint` token. This matches the ledger-row spec's existing assertion that the weekday must never carry `--alp-faint`.

File: frontend/src/app/modules/logging/history/week-aggregate.helpers.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+// Per-week ledger summary for the LOG BOOK history surface (Slice 4 PR-C,
+// spec § 4.2). Computed over a week group's logs (see week-grouping.helpers)
+// and rendered as a single right-aligned line beside the "Week of …" header.
```
[the remark this answers is not in the document]
Comment (leehopper) [reply frontend/src/app/modules/logging/history/week-aggregate.helpers.ts]: Fixed — stripped the PR/slice planning references and source-file cross-references from all four flagged comments, keeping only the self-contained rationale.

File: frontend/src/app/modules/logging/history/week-aggregate.helpers.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+ * Formats a {@link WeekAggregate} as the ledger's week-summary line, e.g.
+ * `"15.2 km · 3 RUNS · 1 SKIP"` (source stays natural-case; the caller's
+ * `uppercase` CSS class is what renders it all-caps). The skip clause is
+ * appended AFTER the run clause only when at least one skip occurred
+ * (spec § 9 #2); a week with no skips renders just the distance + run count.
+ */
+export const formatWeekAggregate = (agg: WeekAggregate, units: PreferredUnits): string => {
+  const km = formatDistanceMeters(agg.distanceMeters, units) ?? '0.0 km'
+  const base = `${km} · ${agg.runCount} RUN${agg.runCount === 1 ? '' : 'S'}`
+  return agg.skipCount > 0 ? `${base} · ${agg.skipCount} SKIP` : base
```
[the remark this answers is not in the document]
Comment (leehopper) [reply frontend/src/app/modules/logging/history/week-aggregate.helpers.ts:53]: Fixed — `formatWeekAggregate`'s zero-distance fallback now uses `distanceUnitLabel(units)` so an all-skipped week under a Miles preference renders "0.0 mi" instead of "0.0 km". Also added a test covering the zero-distance-under-Miles case.

File: frontend/src/app/modules/logging/history/week-aggregate.helpers.ts (hunk 1 above)
Comment (leehopper) [reply frontend/src/app/modules/logging/history/week-aggregate.helpers.ts]: Fixed — same root cause CodeRabbit flagged independently. `formatWeekAggregate` now uses `distanceUnitLabel(units)` for the zero-distance fallback, and added Miles zero-distance test coverage.

**Comment by leehopper** [comment]
@coderabbitai full review
